version: '3'

services:
  db:
    image: postgres:9.6.5
    volumes:
      - postgres_data:/var/lib/postgresql/data/
    healthcheck:
      test: "pg_isready -h localhost -p 5432 -q -U postgres"
      interval: 3s
      timeout: 5s
      retries: 5

  redis:
    image: "redis:alpine"

  nginx:
    image: nginx:1.15.0
    depends_on:
      - django_asgi
      - django_wsgi
    ports:
      - "80:80"
      - "443:443"
    volumes:
      - ./nginx:/etc/nginx/conf.d
      - ./vdx_id/staticfiles:/static
      - ./vdx_id_data:/static/sys_data
      - ./data/certbot/conf:/etc/letsencrypt
      - ./data/certbot/www:/var/www/certbot
    command: "/bin/sh -c 'while :; do sleep 6h & wait $${!}; nginx -s reload; done & nginx -g \"daemon off;\"'"

  django_wsgi:
    container_name: django_wsgi
    image: vdx_identify:core
    build: .
    command: uwsgi --log-master --http=0.0.0.0:9000 -b 32768 --module=vdx_id.wsgi:application --py-autoreload=1
    volumes:
      - ./vdx_id:/opt/vdx_id
      - ./vdx_id_data:/data
    links:
      - redis
    depends_on:
      - db
      - redis
    expose:
      - 9000
    ports:
      - "9999:9999"
      
  django_asgi:
    container_name: django_asgi
    image: vdx_identify:core
    build: .
    command: daphne -b 0.0.0.0 -p 8000 vdx_id.asgi:application
    volumes:
      - ./vdx_id:/opt/vdx_id
      - ./vdx_id_data:/data
    links:
      - redis
    depends_on:
      - db
      - redis
    expose:
      - 8000

volumes:
  postgres_data:
